js/coinsbit.js

```diff
@@ -0,0 +1,799 @@
+            'low': this.safeNumber (tickerFinal, 'low'),
+            'bid': this.safeNumber (tickerFinal, 'bid'),
+            'bidVolume': undefined,
+            'ask': this.safeNumber (tickerFinal, 'ask'),
+            'askVolume': undefined,
+            'vwap': undefined,
+            'open': this.safeNumber (tickerFinal, 'open'),
+            'close': last,
+            'last': last,
+            'previousClose': undefined,
+            'change': undefined, // They do have 'change' property, but atm,  for some reason, they always return "0 | -0 | -1 | -2" strange values. So, skipping for now.
+            'percentage': undefined,
+            'average': undefined,
+            'baseVolume': this.safeNumber (tickerFinal, 'volume'),
+            'quoteVolume': this.safeNumber (tickerFinal, 'deal'),
+            'info': ticker,
+        }, market);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'],
+        };
+        let orderbook = {};
+        if (limit !== undefined) {
+            request['limit'] = Math.min (1000, limit); // this parameter is not required, default 50 will be returned
+        }
+        const type = this.safeString (params, 'type', 'depth/result'); // <<< TODO >> maybe a better way to set default unified param name (same param can be used for binance fetchBidsAsks too, if that will get integrated inside binance's fetchOrderbook )
+        const method = this.safeString (this.options, 'fetchOrderBookMethod', type);
+        if (method === 'depth/result') {
+            const response = await this.publicGetDepthResult (this.extend (request, params));
+            //     {
+            //         asks: [
+            //             [  "46809.06069986", "2.651"  ],
+            //             [  "46819.80796146", "2.6401" ],
+            //             ...
+            //         ],
+            //         bids: [
+            //             [  "46799.39054516",  "2.702" ],
+            //             [  "46801.25196599",  "0.4"   ],
+            //             ...
+            //         ]
+            //     }
+            const priceKey = 0;
+            const amountKey = 1;
+            orderbook = this.parseOrderBook (response, symbol, undefined, 'bids', 'asks', priceKey, amountKey);
+        } else if (method === 'book') {
+            request['offset'] = this.safeString (params, 'offset', 0);
+            // ATM, they need to call separately (unfortunately)
+            request['side'] = this.safeString (params, 'side', 'buy');
+            const requestBids = this.publicGetBook (this.extend (request, params));
+            request['side'] = this.safeString (params, 'side', 'sell');
+            const requestAsks = this.publicGetBook (this.extend (request, params));
+            const responses = await Promise.all ([requestBids, requestAsks]);
```

Comment:
We cannot do `Promise.all` in derived classes (no other exchange will have a mention of `Promise` anywhere around the lib). We should do those requests sequentially, or require a side from the user.